library/src/main/java/com/pokegoapi/main/RequestHandler.java

```diff
@@ -313,4 +304,94 @@ public void run() {
 			}
 		}
 	}
+
+	private void setSignature(AuthTicket authTicket, RequestEnvelope.Builder builder, ServerRequest[] serverRequests) {
+		if (authTicket == null) {
+			System.out.println("Ticket == null");
+			return;
+		}
+		byte[] uk22 = new byte[32];
+		new Random().nextBytes(uk22);
+
+		long curTime = api.currentTimeMillis();
+
+		byte[] authTicketBA = authTicket.toByteArray();
+
+		SignatureOuterClass.Signature.Builder sigBuilder = SignatureOuterClass.Signature.newBuilder()
+				.setLocationHash1(getLocationHash1(authTicketBA, builder))
+				.setLocationHash2(getLocationHash2(builder))
+				.setUnk22(ByteString.copyFrom(uk22))
+				.setTimestamp(api.currentTimeMillis())
+				.setTimestampSinceStart(curTime - api.startTime);
+
+
+		for (ServerRequest serverRequest : serverRequests) {
+			byte[] request = serverRequest.getRequest().toByteArray();
+			sigBuilder.addRequestHash(getRequestHash(authTicketBA, request));
+		}
+
+		// TODO: Call encrypt function on this
+		byte[] uk2 = sigBuilder.build().toByteArray();
+		byte[] iv = new byte[32];
+		new Random().nextBytes(iv);
+		byte[] encrypted = Crypto.encrypt(uk2, iv).toByteBuffer().array();
+		System.out.println(uk2);
+		System.out.println(encrypted);
+		Unknown6OuterClass.Unknown6.newBuilder()
+				.setRequestType(6)
+				.setUnknown2(Unknown6OuterClass.Unknown6.Unknown2.newBuilder().setUnknown1(ByteString.copyFrom(encrypted)));
+	}
+
+	private static byte[] getBytes(double d) {
+		long x = Double.doubleToRawLongBits(d);
+		return new byte[]{
+				(byte) (x >>> 56),
+				(byte) (x >>> 48),
+				(byte) (x >>> 40),
+				(byte) (x >>> 32),
+				(byte) (x >>> 24),
+				(byte) (x >>> 16),
+				(byte) (x >>> 8),
+				(byte) x
+		};
+	}
+
+	private int getLocationHash1(byte[] authTicket, RequestEnvelope.Builder builder) {
+		XXHashFactory factory = XXHashFactory.fastestInstance();
+		StreamingXXHash32 xx32 = factory.newStreamingHash32(0x1B845238);
+		xx32.update(authTicket, 0, authTicket.length);
+		xx32 = factory.newStreamingHash32(xx32.getValue());
+		byte[] bytes = new byte[8 * 3];
+
+		System.arraycopy(getBytes(api.getLatitude()), 0, bytes, 0, 8);
+		System.arraycopy(getBytes(api.getLongitude()), 0, bytes, 8, 8);
```

Comment:
As it is sent via protobuf I think that shouldn't matter, as it's the same memory representation

http://ideone.com/Y0DLIo